Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/ModularViewportCameraController.cpp

```diff
@@ -285,18 +291,24 @@ namespace AtomToolsFramework
             if (animationTime >= 1.0f)
             {
                 m_cameraMode = CameraMode::Control;
+                m_cameraAnimation.reset();
             }
         }
 
         m_updatingTransformInternally = false;
     }
 
-    bool ModularViewportCameraControllerInstance::InterpolateToTransform(const AZ::Transform& worldFromLocal)
+    bool ModularViewportCameraControllerInstance::InterpolateToTransform(const AZ::Transform& worldFromLocal, const float duration)
     {
-        if (!IsInterpolating())
+        const auto& currentCameraTransform = CombinedCameraTransform();
+
+        // ensure the transform we're interpolating to isn't the same as our current transform
+        // and the transform we're setting isn't the same as one previously set
+        if (!currentCameraTransform.IsClose(worldFromLocal) &&
+            (!m_cameraAnimation.has_value() || !worldFromLocal.IsClose(m_cameraAnimation->m_transformEnd)))
         {
             m_cameraMode = CameraMode::Animation;
-            m_cameraAnimation = CameraAnimation{ CombinedCameraTransform(), worldFromLocal, 0.0f };
+            m_cameraAnimation = CameraAnimation{ currentCameraTransform, worldFromLocal, 0.0f, duration };
```

Comment:
If the new transform is close, shouldn't it still set it as the new transform immediately instead of ignoring it? It seems like this could lead to an edge case behavior where tiny intentional changes just get dropped.

Reply:
I do not believe so no. If you're attempting to interpolate to where you already are, before what would happen is the 'animation' would play, but you don't move (as it's in the same place) which feels really weird if you then try and move the camera (this is easy to do when quickly moving between view bookmarks).

We use an optional so `cameraAnimation` can be fully 'unset' when not in use - this is to stop users spamming 'Go to position' when initially first moving to the new location. I did add several tests to replicate some of the edge cases, let me know if there's another you think is worth adding (we can hop on a call to discuss if needed). Thanks!

Reply:
I guess what I was thinking was more of this:
1. If an animation is currently playing, and a new endpoint is requested that IsClose() to the current endpoint, then the current one should get adjusted to the new endpoint without restarting the animation.
2. If no animation is currently playing, and a new endpoint is requested that IsClose() to the current location, it would just immediately jump to the new value.
This way it would preserve the new request, even if it's really close to the previous one, without introducing the restarting animation problem that you're describing. But it's also super super nitpicky and not something that I was overly concerned about, just figured it was worth asking.

Reply:
I guess my feeling with point 1 is, if `IsClose()` is true, then it'd be equivalent to (as far is the user is concerned) to updating it and not starting the animation again, so I personally don't think it's necessary to update it.

Also with point 2, if the requested location is `IsClose` to the current position, there's no point in jumping there as you're already there, so we just skip doing that too 😝

So I don't think there is an edge case that would result in an observable negative outcome, we're just talking about slightly different approaches reaching the same end.